Feature/cleanup

Fangmeister opened the pull request:
Javadocs added to Vet
Cleanup of redundant functions

Fangmeister committed:
WIP VetShop javadocs

- deleted OwnerName enum as it was redundant
- deleted getFurType() in Animal
- deleted getStayDuration(), getCatName(), getWeight(), getCatBreed() in Cat as all are public parameters; replaced functions with .stayDuration etc. in VetShop
- deleted getStayDuration(), getDogName(), getWeight(), getDogBreed() in Dog as all are public parameters; replaced functions with .stayDuration etc. in VetShop
- deleted addAnimal() and removeAnimal() from the Vet Class as these functions are not invoked in the current program
- cleaned up old comments
- removed unnecessary imports for each java file
- added javadocs to all files in the animal package
- added javadocs to all Clerk, Employee and Vet in the veterinary package

Fangmeister committed:
Final javadoc

- added javadoc to VetShop
- added input.close() to quit funtionality
- added javadoc to DayHospital

dvsNefilim commented on src/DayHospital.java:
```diff
@@ -73,37 +91,65 @@ public static void main(String[] args) {
 	}
 	
+	/**
+	 * Returns a random float between the floats 1.1 and 0.2
```
This is incorrect, and the wording is weird. 0.2 and 1.3. (by wording is weird I mean that you would normally start with the lowest value)

Fangmeister replied:
changed to "floats between 0.2 and 1.3"

dvsNefilim commented on src/DayHospital.java:
```diff
@@ -73,37 +91,65 @@ public static void main(String[] args) {
 	}
 
+	/**
+	 * Returns a random float between the integers 3 and 20
```
This is incorrect 3 and 23

Fangmeister replied:
changed actual random to Math.random()*17 + 3 thereby making the original javadoc accurate

dvsNefilim commented on src/DayHospital.java:
```diff
@@ -73,37 +91,65 @@ public static void main(String[] args) {
+			 *  - Weight
+			 * The animals contained in the kennels array should be the same as what is assigned to the vets.
+			 * Note that vets are assigned animals by preference of least number of animals per vet first, so the kennel arrays and vet arrays will not correspond to each other.
+			 * Rather the unique animal instances in the kennels array should be same as the unique animal instances in the vet arrays.
```
This wording is confusing. Don't have "Rather", make it clear the first time.

Fangmeister replied:
Changed wording

dvsNefilim commented on src/DayHospital.java:
```diff
@@ -20,44 +17,65 @@ public static void main(String[] args) {
 		System.out.println("Enter your input please: ");
 		
+		/**
+		 * A new instance of VetShop, petInterCare, is created
```
If you are going to make a 1 line comment just use // otherwise you are just bloating the code

Fangmeister replied:
changed to //

dvsNefilim commented on src/DayHospital.java:
```diff
@@ -20,44 +17,65 @@ public static void main(String[] args) {
-		
-		while (!temp.equalsIgnoreCase(quit)) {
+		/**
+		 * A while loop that runs as long as the console input is not equal to q or Q
```
//

Fangmeister replied:
done

dvsNefilim commented on src/DayHospital.java:
```diff
@@ -20,44 +17,65 @@ public static void main(String[] args) {
+			 * A random number of animals (between 1 and 10) are generated. 
+			 * There is a text output of the number of new animals generated.
+			 * Each animal is then processed as follows:
+			 * 	- the species of the animal is randomly generated via a boolean (if true, animal is a new Dog; if false, animal is a new Cat).
```
tab misalignment

Fangmeister replied:
it's fine in the code

dvsNefilim commented on src/animals/Animal.java:
```diff
@@ -43,48 +52,79 @@
 	
-	
+	/**
+	 * Create a new Animal instance. The Animal class is a super class of the sub classes Cat and Dog.
```
tab alignment is out of whack

Fangmeister replied:
it's fine in the code

dvsNefilim commented on src/animals/Cat.java:
```diff
@@ -1,10 +1,13 @@
 public class Cat extends Animal {
 	
+	/**
+	 * CatBreed enum catBreed is defined for the Cat Class. It is used in the Cat constructor.
```
the comment needs to go above the relevant piece of code. Not in one block. Move the cat name description to above catName. You are also describing the CatBreed enum here, not the variable catBreed. The enum description should go with the enum definition.

Fangmeister replied:
changed for enum description to be above catBreed; catName description above catName[]

dvsNefilim commented on src/animals/Cat.java:
```diff
@@ -30,27 +33,25 @@
 	
+	/**
+	 * Creates an instance of the Cat class, which is a subclass of the Animal class. Cat has one additional parameter to Animal: Cat Breed.
+	 * @param catNameIn				The name of the cat
```
tabbing

Fangmeister replied:
no problem in code

dvsNefilim commented on src/animals/Cat.java:
```diff
@@ -30,27 +33,25 @@
+	 * @return The catName list variable at the random index generated
+	 */
 	public static String getRandomCatName() {
 	Random random = new Random();
```
code is misaligned

Fangmeister replied:
fixed

dvsNefilim commented on src/animals/Dog.java:
```diff
@@ -1,11 +1,14 @@
+   * dogName is a string array which contains the list of possible names for dogs.
+   */
+  public DogBreed dogBreed;
 	public static String[] dogName = {
```
tabbing and same comment as in cat where the comment needs to be located above the relevant code.

Fangmeister replied:
fixed as in the Cats doc

dvsNefilim commented on src/animals/Dog.java:
```diff
@@ -29,36 +32,28 @@
 
+	/**
+	 * Creates an instance of the Dog class, which is a subclass of the Animal class. Dog has one additional parameter to Animal: Dog Breed.
+	 * @param dogNameIn      The name of the dog
```
tabbing

Fangmeister replied:
no problem in code

dvsNefilim commented on src/veterinary/Clerk.java:
```diff
@@ -2,6 +2,12 @@
 	
+	/**
+	 * Creates an instance of the Clerk Class, which is a sub class of the Employee Class. The parameters for Clerk and Employee are the same.
+	 * @param nameIn				The name of the clerk
```
tabbing

Fangmeister replied:
no problem in code

dvsNefilim commented on src/veterinary/Vet.java:
```diff
@@ -4,20 +4,22 @@
 import animals.Animal;
 
 public class Vet extends Employee {
+	/**
```
double start is reserved for functions and classes (i.e. "/**")

Fangmeister replied:
fixed

dvsNefilim commented on src/veterinary/VetShop.java:
```diff
@@ -1,14 +1,21 @@
 import java.util.Random;
 
 public class VetShop {
+	/**
```
double star not applicable here. Move comments to be above the relevant code

Fangmeister replied:
fixed to one star

Fangmeister replied:
moved comments to be above relevant code

Fangmeister committed:
Addressing pull request comments

- updated comments
- changed double star to one star for non functions
- moved comments to above relevant code
- changed one line java docs to comments
- changed wording for clarity
